cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/ReplicationThrottleHelper.java

```diff
@@ -390,4 +396,68 @@ static boolean configsEqual(Config configs, Map<String, String> expectedValues)
     }
     return true;
   }
+
+  private enum LogConfig {
+    LEADER_REPLICATION_THROTTLED_REPLICAS_CONFIG,
+    FOLLOWER_REPLICATION_THROTTLED_REPLICAS_CONFIG,
+  }
+
+  /**
+   * Starting with Kafka 3.5.0, the location of the "LogConfig" class is "org.apache.kafka.storage.internals.log.LogConfig"
+   * and provides new constant fields in place of some methods of the original class.
+   *
+   *   - LogConfig class in Kafka 3.5+: org.apache.kafka.storage.internals.log.LogConfig
+   *   - LogConfig class in Kafka 3.4-: kafka.log.LogConfig
+   **
+   * The older LogConfig class does not work with the newer versions of Kafka. Therefore, if the new class exists, we use it and if
+   * it doesn't exist we will fall back on the older one.
+   *
+   * Once CC supports only 3.5.0 and newer, we can clean this up and use the LogConfig class from
+   * `org.apache.kafka.storage.internals.log.LogConfig`all the time.
+   * @param config LogConfig config name
+   * @return LogConfig config name in the format of a Kafka configuration property
+   */
+  private static String getLogConfig(LogConfig config) {
+    Class<?> logConfigClass;
+
+    try {
+      // First we try to get the LogConfig class for Kafka 3.5+
+      logConfigClass = Class.forName(LOG_CONFIG_IN_KAFKA_3_5_AND_LATER);
+      LOG.info("Found class {} for Kafka 3.5 and newer.", LOG_CONFIG_IN_KAFKA_3_5_AND_LATER);
+    } catch (ClassNotFoundException e) {
+      LOG.info("Class {} not found. We are probably on Kafka 3.4 or older.", LOG_CONFIG_IN_KAFKA_3_5_AND_LATER);
+
+      // We did not find the LogConfig class from Kafka 3.5+.
+      // So we are probably on older Kafka version => we will try the older class for Kafka 3.4-.
+      try {
+        logConfigClass = Class.forName(LOG_CONFIG_IN_KAFKA_3_4_AND_EARLIER);
+        LOG.info("Found class {} for Kafka 3.4 and earlier.", LOG_CONFIG_IN_KAFKA_3_4_AND_EARLIER);
+      } catch (ClassNotFoundException ee) {
+          // No class was found for any Kafka version => we should fail
+          throw new RuntimeException("Failed to find LogConfig class", ee);
+        }
+    }
+
+    try {
```

Comment:
To me, it seems to better put move L442~L443 to L426, and move L451~L458 to L434. E.g. in general, there are only 2 cases: 3.4 or before, and 3.5 or after. We should merge the logic into only these 2 branches.

Reply:
@CCisGG  Yeah you are right, that does sound better.  Just refactored code based on the feedback, let me now what you think!